src/test/java/org/wso2/lsp4intellij/utils/FileUtilsTest.java

```diff
@@ -0,0 +1,200 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.wso2.lsp4intellij.utils;
+
+import com.intellij.openapi.editor.Document;
+import com.intellij.openapi.editor.Editor;
+import com.intellij.openapi.fileEditor.FileDocumentManager;
+import com.intellij.openapi.fileEditor.FileEditor;
+import com.intellij.openapi.fileEditor.FileEditorManager;
+import com.intellij.openapi.fileEditor.TextEditor;
+import com.intellij.openapi.fileEditor.ex.FileEditorManagerEx;
+import com.intellij.openapi.project.Project;
+import com.intellij.openapi.vfs.LocalFileSystem;
+import com.intellij.openapi.vfs.VirtualFile;
+import com.intellij.testFramework.BinaryLightVirtualFile;
+import com.intellij.testFramework.LightVirtualFile;
+import java.io.File;
+import java.net.URISyntaxException;
+import org.junit.Assert;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mockito;
+import org.powermock.api.mockito.PowerMockito;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+import org.powermock.modules.junit4.PowerMockRunner;
+
+@RunWith(PowerMockRunner.class)
+public class FileUtilsTest {
+
+  @PrepareForTest(FileEditorManager.class)
+  @Test
+  public void testEditorFromVirtualFile() {
+    VirtualFile file = PowerMockito.mock(VirtualFile.class);
+    Project project = PowerMockito.mock(Project.class);
+    
+    Editor editor = PowerMockito.mock(Editor.class);
+    TextEditor textEditor = PowerMockito.mock(TextEditor.class);
+    PowerMockito.when(textEditor.getEditor()).thenReturn(editor);
+
+    FileEditorManagerEx fileEditorManagerEx = PowerMockito.mock(FileEditorManagerEx.class);
+    PowerMockito.mockStatic(FileEditorManager.class);
+    PowerMockito.when(fileEditorManagerEx.getAllEditors(file))
+        .thenReturn(new FileEditor[]{textEditor})
+        .thenReturn(new FileEditor[0]);
+    PowerMockito.when(FileEditorManager.getInstance(project)).thenReturn(fileEditorManagerEx);
+
+    Assert.assertEquals(editor, FileUtils.editorFromVirtualFile(file, project));
+
+    Assert.assertNull(FileUtils.editorFromVirtualFile(file, project));
+  }
+
+  @PrepareForTest(LocalFileSystem.class)
+  @Test
+  public void testVirtualFileFromURI() {
+    LocalFileSystem localFileSystem = PowerMockito.mock(LocalFileSystem.class);
+    PowerMockito.mockStatic(LocalFileSystem.class);
+    PowerMockito.when(LocalFileSystem.getInstance()).thenReturn(localFileSystem);
+    PowerMockito.when(LocalFileSystem.getInstance().findFileByIoFile(Mockito.any()))
+        .thenReturn(new BinaryLightVirtualFile());
+
+    Assert.assertEquals(new BinaryLightVirtualFile().toString(),
+        FileUtils.virtualFileFromURI("file://foobar").toString());
+  }
+
+  @PrepareForTest(LocalFileSystem.class)
+  @Test
+  public void testVirtualFileFromURINull() {
+    PowerMockito.mockStatic(LocalFileSystem.class);
+    PowerMockito.when(LocalFileSystem.getInstance()).thenThrow(URISyntaxException.class);
```

Comment:
Why is this set up to throw an exception which doesn't get thrown?

Reply:
this is done to reach a line in a try catch statement which only catches this specific exception.